controlplane/eks/api/v1beta2/types.go

```diff
@@ -279,3 +277,17 @@ type OIDCIdentityProviderConfig struct {
 	// +optional
 	Tags infrav1.Tags `json:"tags,omitempty"`
 }
+
+// PodIdentityAssociation represents an association between a Kubernetes Service Account in a namespace, and an AWS IAM role which has "pods.eks.amazonaws.com" as a service principal.
+type PodIdentityAssociation struct {
+	// ServiceAccountName is the name of the kubernetes Service Account within the namespace
+	// +kubebuilder:validation:Required
+	ServiceAccountName string `json:"serviceAccountName"`
+	// ServiceAccountNamespace is the kubernetes namespace, which the kubernetes Service Account resides in. Defaults to "default" namespace.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default=default
+	ServiceAccountNamespace string `json:"serviceAccountNamespace"`
+	// RoleARN is the ARN of an IAM role which the Service Account can assume.
+	// +kubebuilder:validation:Required
+	RoleARN string `json:"serviceAccountRoleARN,omitempty"`
```

Comment:
The goal here was to be overtly explicit as I have found that our customers get confused on that sometimes. Nothing that docs cannot fix however

Reply:
In thinking more about this, I remembered that the kubernetes Pod spec switched from `serviceAccount` to `serviceAccountName` a few years back. 
I do err on the side of clarity for maintainability's sake when it comes to naming, but would definitely not go against whatever is considered standard. Do you have a reference for the naming convention to help me decide?

Reply:
In thinking about this more, whilst the medium is a CRD, the keys are in reference to an AWS object. I would expect a PodIdentityAssociation.Name to refer to the name of an AWS object when using the AWS API

Reply:
`Pod.spec.serviceAccountName` is a singular field that won't change. It doesn't allow referring to another namespace on purpose.

The 3 fields you're adding will always be filled and none can be optional or defaulted (even the namespace), so a common prefix is unusual for newly-designed CRDs. See for example `Ingress.spec.defaultBackend.service.{name,port}` or Gateway API's `Gateway.[...].listeners[*].{name,protocol,port}`. Core APIs such as `Pod` aren't great examples since they can't be evolved without lots of breaking changes, and also a pod spec mostly only refers to names and doesn't provide many comparable examples here.

Therefore, here's my suggestion (dedenting `roleARN` as compared to my initial comment, since it's not referring to a service account):

```
  podIdentityAssociations:
    - serviceAccount:
        name: myserviceaccount
        namespace: default
      roleARN: arn:aws:iam::012345678901:role/capi-test-role
```

Reply:
There is definitely a back and forth on the semantics on this. I personally dont see this as being a blocking change, and moreso a semantic preference.

Reply:
At least the JSON field name and struct name should be aligned, since currently they're not lowercase-equal which could be confusing:

```suggestion
	// ServiceAccountName is the name of the kubernetes Service Account within the namespace
	// +kubebuilder:validation:Required
	ServiceAccountName string `json:"serviceAccountName"`
	// ServiceAccountNamespace is the kubernetes namespace, which the kubernetes Service Account resides in. Defaults to "default" namespace.
	// +kubebuilder:validation:Required
	// +kubebuilder:default=default
	ServiceAccountNamespace string `json:"serviceAccountNamespace"`
	// RoleARN is the ARN of an IAM role which the Service Account can assume.
	// +kubebuilder:validation:Required
	ServiceAccountRoleARN string `json:"serviceAccountRoleARN,omitempty"`
```

Or the other way around (`json:"roleARN"`) – doesn't matter.

Then the documented examples in `docs/book/src/topics/eks/pod-identity-associations.md` must be converted to the final naming choice.

Reply:
In this case, the roleARN is an AWS resource, not a service account, so the naming shouldnt be consistent. The suggested shouldn't be committed imo